07.ls_object/file_detail.rb

```diff
@@ -3,46 +3,47 @@
 require 'etc'
 
 class FileDetail
-  attr_reader :file_name, :file_stat
+  attr_reader :file_name
+
+  TYPES = { 'file' => '-', 'directory' => 'd', 'link' => 'l' }.freeze
+  PERMISSIONS = { '0' => '---', '1' => '--X', '2' => '-w-', '3' => '-wx', '4' => 'r--', '5' => 'r-x', '6' => 'rw-', '7' => 'rwx' }.freeze
 
   def initialize(file_name)
     @file_name = file_name
     @file_stat = File.stat(file_name)
   end
 
+  def blocks
+    @file_stat.blocks 
+  end
+
   def ftype
-    ftype_key = File.ftype(file_name)
-    types = { 'file' => '-', 'directory' => 'd', 'link' => 'l' }
-    types[ftype_key]
+    ftype_key = File.ftype(@file_name)
+    TYPES[ftype_key]
   end
 
   def permissions
-    permission_numbers = file_stat.mode.to_s(8).slice(-3, 3).chars
-    permission_table = { '0' => '---', '1' => '--X', '2' => '-w-', '3' => '-wx', '4' => 'r--', '5' => 'r-x', '6' => 'rw-', '7' => 'rwx' }
-    permission_numbers.map { |num| permission_table[num] }.join
+    permission_numbers = @file_stat.mode.to_s(8).slice(-3, 3).chars
+    permission_numbers.map { |num| PERMISSIONS[num] }.join
```

Comment:
```suggestion
    permission_number = @file_stat.mode.to_s(8).slice(-3, 3)
    permission_number.gsub(/[0-7]/, PERMISSIONS)
```

かなり高等テクニックですがこんな書き方もできます。

Reply:
「数字の文字列」を１文字ずつ「パーミッションの文字列」に置き換えるんですね👀

１歩ずつ手順踏まないと理解が追いついていなくて、とても回りくどい方法にしてました😅

ダイレクトに変換できるので、こちらの方がスッキリしますね🙌🏻